### PR TITLE
Feature/check icon on address screens

### PR DESCRIPTION
**Pull request by xidealo**

### CODE REVIEW

File: shared/src/commonMain/kotlin/com/bunbeauty/shared/presentation/create_order/CreateOrderViewModel.kt [hunk 1]
```diff
@@ -66,7 +67,11 @@ class CreateOrderViewModel(
         val event = if (addressUiList.isEmpty()) {
             OrderCreationState.Event.OpenCreateAddressEvent
         } else {
-            OrderCreationState.Event.ShowUserAddressListEvent(addressUiList)
+            OrderCreationState.Event.ShowUserAddressListEvent(
+                addressList = addressUiList,
+                selectedUserAddressUuid = orderCreationState.value.deliveryAddress?.uuid
+                    ?: Constants.NOT_SELECTED
```
Comment (MaxAstin) [inline shared/src/commonMain/kotlin/com/bunbeauty/shared/presentation/create_order/CreateOrderViewModel.kt:73]: Why we need this constant? Why not use `null`?

File: shared/src/commonMain/kotlin/com/bunbeauty/shared/presentation/create_order/CreateOrderViewModel.kt [hunk 2]
```diff
@@ -82,7 +87,9 @@ class CreateOrderViewModel(
 
     fun onCafeAddressClicked() {
         val event = OrderCreationState.Event.ShowCafeAddressListEvent(
-            orderCreationData.value.cafeList.map(CafeAddressMapper::toCafeAddressItem)
+            orderCreationData.value.cafeList.map(CafeAddressMapper::toCafeAddressItem),
+            selectedCafeAddress = orderCreationState.value.pickupAddress
+                ?: Constants.NOT_SELECTED
```
Comment (MaxAstin) [inline shared/src/commonMain/kotlin/com/bunbeauty/shared/presentation/create_order/CreateOrderViewModel.kt:92]: `null`?

File: shared/src/commonMain/kotlin/com/bunbeauty/shared/presentation/settings/SettingsViewModel.kt [hunk 3]
```diff
@@ -60,7 +62,10 @@ class SettingsViewModel(
 
     fun onCityClicked() {
         mutableSettingsState.update { settingsState ->
-            settingsState + SettingsState.Event.ShowCityListEvent(settingsState.cityList)
+            settingsState + SettingsState.Event.ShowCityListEvent(
+                cityList = settingsState.cityList,
+                selectedCityUuid = settingsState.selectedCity?.uuid ?: NOT_SELECTED
```
Comment (MaxAstin) [inline shared/src/commonMain/kotlin/com/bunbeauty/shared/presentation/settings/SettingsViewModel.kt:67]: `null`?

**Review by MaxAstin — CHANGES_REQUESTED** [review]

File: shared/src/commonMain/kotlin/com/bunbeauty/shared/presentation/settings/SettingsViewModel.kt (hunk 3 above)
Comment (xidealo) [reply shared/src/commonMain/kotlin/com/bunbeauty/shared/presentation/settings/SettingsViewModel.kt:67]: +

File: shared/src/commonMain/kotlin/com/bunbeauty/shared/presentation/create_order/CreateOrderViewModel.kt (hunk 2 above)
Comment (xidealo) [reply shared/src/commonMain/kotlin/com/bunbeauty/shared/presentation/create_order/CreateOrderViewModel.kt:92]: +

File: shared/src/commonMain/kotlin/com/bunbeauty/shared/presentation/create_order/CreateOrderViewModel.kt (hunk 1 above)
Comment (xidealo) [reply shared/src/commonMain/kotlin/com/bunbeauty/shared/presentation/create_order/CreateOrderViewModel.kt:73]: +

File: app/src/main/java/com/bunbeauty/papakarlo/feature/address/ui/AddressItem.kt [hunk 4]
```diff
@@ -26,14 +36,29 @@ fun AddressItem(
...
-            text = address,
-            style = FoodDeliveryTheme.typography.bodyLarge,
-            color = FoodDeliveryTheme.colors.onSurface
-        )
+        Row {
+            Text(
+                modifier = Modifier
+                    .weight(1f)
+                    .padding(vertical = FoodDeliveryTheme.dimensions.mediumSpace)
+                    .padding(start = FoodDeliveryTheme.dimensions.mediumSpace)
```
[the remark this answers is not in the document]
Comment (xidealo) [reply app/src/main/java/com/bunbeauty/papakarlo/feature/address/ui/AddressItem.kt:44]: +

File: app/src/main/java/com/bunbeauty/papakarlo/feature/address/ui/AddressItem.kt [hunk 5]
```diff
@@ -26,14 +36,38 @@ fun AddressItem(
...
             modifier = Modifier
-                .fillMaxWidth()
-                .padding(FoodDeliveryTheme.dimensions.mediumSpace),
-            text = address,
-            style = FoodDeliveryTheme.typography.bodyLarge,
-            color = FoodDeliveryTheme.colors.onSurface
-        )
+                .padding(end = 16.dp)
+                .padding(vertical = FoodDeliveryTheme.dimensions.mediumSpace)
+                .padding(start = FoodDeliveryTheme.dimensions.mediumSpace)
```
Comment (MaxAstin) [inline app/src/main/java/com/bunbeauty/papakarlo/feature/address/ui/AddressItem.kt:43]: `mediumSpace` = 16.dp, So, I suggested  `.padding(all = 16.dp)`

File: app/src/main/java/com/bunbeauty/papakarlo/feature/address/ui/AddressItem.kt [hunk 6]
```diff
@@ -26,14 +36,38 @@ fun AddressItem(
...
+        ) {
+            Text(
+                modifier = Modifier
+                    .weight(1f)
+                    .padding(
+                        end =
```
Comment (MaxAstin) [inline app/src/main/java/com/bunbeauty/papakarlo/feature/address/ui/AddressItem.kt:49]: Why? I think it should always be 0.dp

File: app/src/main/java/com/bunbeauty/papakarlo/feature/address/ui/AddressItem.kt [hunk 7]
```diff
@@ -26,14 +36,38 @@ fun AddressItem(
...
+                        } else {
+                            0.dp
+                        }
+                    ),
+                text = address,
+                style = FoodDeliveryTheme.typography.bodyLarge,
+                color = FoodDeliveryTheme.colors.onSurface
+            )
+            if (isSelected) {
+                Icon(
```
Comment (MaxAstin) [inline app/src/main/java/com/bunbeauty/papakarlo/feature/address/ui/AddressItem.kt:61]: And here you can add `padding` or `Spacer`

**Review by MaxAstin — CHANGES_REQUESTED** [review]

**Review by MaxAstin — APPROVED** [review]

**Pull request merged**